app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    class UsersController < ApplicationController
+      # POST /api/v1/users.json (the json format is required for an explicit API)
+      # Expects:
+      #     params with the fallowing format: { user: { :name, :email, :password, :password_confirmation } }
+      # Does:
+      #     - Filters and validates the recieved params( throws ActionController::ParameterMissing for missed params).
+      #     - Creates and saves a new user record in Users table with the provided parameters.
+      #     - Returns json with the fallowig format : { msg: String, data: Array[serializable objects]
+      #      ,errors: Array[String] }
+      #     and status_code: 201 (created) | 400 (bad request).
+      #         msg = response_msgs[:success] and errors = [] for successful save to db with status_code = 201.
+      #         msg = response_msgs[:failed] and errors = user.errors.to_a for unsuccessfuly save to db
+      #         with status_code = 400.
+      def create
+        user = User.new({ provider: 'greenlight' }.merge(user_params)) # TMP fix for presence validation of :provider
+        if user.save
+          render json: {
+            msg: Rails.configuration.response_msgs[:success],
```

Comment:
The more I think about this, the more I feel like this is redundant. Whats the point of this while the status exists?

Reply:
Just a consistent response format because of the lack of one.
It's not final and has to be rectified according to our need.
1. The msg can contain a more explicit status message that may be used on client side.
2. The data is where any data sent back to client side lives.
3. The error is where any error messages flows between the backend -> frontend.

Reply:
What can I do?

Reply:
The `data` and `error` parts are good, my thought is around the `msg` part. `status` already exists so that we can return a ton of different responses based on what happened to the client side, so duplicating it here seems kind of odd